PHPCompatibility/Sniffs/Constants/RemovedConstantsSniff.php

```diff
@@ -2718,6 +2723,35 @@ protected function handleFeature(File $phpcsFile, $stackPtr, array $itemInfo)
         $versionInfo = $this->getVersionInfo($itemArray);
         $isError     = null;
 
+        if ($itemInfo['name'] === 'T_BAD_CHARACTER') {
```

Comment:
Same remark as I made for `NewConstants` - this can be greatly simplified by undoing the change to the version array, moving the logic to below the `isset($isError) === false` condition block and changing it to:

```php
        // T_BAD_CHARACTER is a special case. It was removed in 7.0.0 and re-added in 7.4.0
        // See also PHPCompatibility.Constants.NewConstants
        if ($itemInfo['name'] === 'T_BAD_CHARACTER') {
            if (ScannedCode::shouldRunOnOrBelow('7.3') === true) {
                $message = 'The constant "T_BAD_CHARACTER" is not present in PHP versions 7.0 through 7.3';
                $msgInfo = $this->getMessageInfo($itemInfo['name'], $itemInfo['name'], $versionInfo);

                $phpcsFile->addError($message, $stackPtr, $msgInfo['errorcode'], $msgInfo['data']);
            }

            return;
        }
```

Reply:
Thanks for the suggestion. Upon reflection, this makes sense. I've applied this change and the test suite still passes.